File: app/src/main/java/com/gustu/logbook/login/view/LoginActivity.java
package com.gustu.logbook.login.view;

import androidx.appcompat.app.AppCompatActivity;

import android.app.AlertDialog;
import android.content.Intent;
import android.os.Bundle;
import android.widget.EditText;
import android.widget.Toast;

import com.gustu.logbook.R;
import com.gustu.logbook.login.interfaces.LoginView;
import com.gustu.logbook.login.model.LoginResponse;
import com.gustu.logbook.login.presenter.LoginPresenter;
import com.gustu.logbook.main.view.MainActivity;
import com.gustu.logbook.sharePreferences.SharedPrefUtil;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import dmax.dialog.SpotsDialog;

public class LoginActivity extends AppCompatActivity implements LoginView {
    @BindView(R.id.etUsername)
    EditText ETusername;
    @BindView(R.id.etPassword)
    EditText ETpassword;
    LoginPresenter loginPresenter;
    AlertDialog progressDialog;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);
        ButterKnife.bind(this);
        initPresenter();
        progressDialog = new SpotsDialog(this,R.style.DialogCustom);

    }

    private void initPresenter() {
        loginPresenter = new LoginPresenter(this);
    }

    @OnClick(R.id.btMasuk)
    public void login(){
        String username = ETusername.getText().toString();
        String password = ETpassword.getText().toString();
        if (username.isEmpty() || password.isEmpty()){
            Toast.makeText(LoginActivity.this,"Form Tidak Boleh Kosong",Toast.LENGTH_SHORT).show();
        }
        else {
            progressDialog.show();
            loginPresenter.gotoLogin(username,password);
        }
    }
    @Override
    public void _onLogin(LoginResponse loginResponse) {
        progressDialog.dismiss();
        SharedPrefUtil.saveBoolean("isLogin",true);
        SharedPrefUtil.saveString("nama",loginResponse.getJsonMember0().getMPGNAMAGELAR());
        SharedPrefUtil.saveString("kode_pegawai",loginResponse.getJsonMember0().getMPGKODE());
        SharedPrefUtil.saveString("kode_unit",loginResponse.getJsonMember0().getMSUKODE());
        startActivity(new Intent(LoginActivity.this, MainActivity.class));
        finish();
    }

    @Override
    public void _onFailed() {
        progressDialog.dismiss();
        Toast.makeText(this,"Login Gagal",Toast.LENGTH_SHORT).show();
    }
}

File: app/src/main/java/com/gustu/logbook/login/view/LoginActivity.kt
package com.gustu.logbook.login.view

import androidx.appcompat.app.AppCompatActivity

import android.content.Intent
import android.graphics.Color
import android.os.Build
import android.os.Bundle
import android.view.View
import android.widget.TextView
import android.widget.Toast
import androidx.annotation.RequiresApi
import com.google.android.material.snackbar.Snackbar

import com.gustu.logbook.R
import com.gustu.logbook.login.interfaces.LoginView
import com.gustu.logbook.login.model.LoginResponse
import com.gustu.logbook.login.presenter.LoginPresenter
import com.gustu.logbook.main.view.MainActivity
import com.gustu.logbook.sharePreferences.SharedPrefUtil
import dmax.dialog.SpotsDialog
import kotlinx.android.synthetic.main.activity_login.*

class LoginActivity : AppCompatActivity(), LoginView {
    internal lateinit var loginPresenter: LoginPresenter
    internal lateinit var progressDialog: SpotsDialog
    @RequiresApi(Build.VERSION_CODES.O)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)
        initPresenter()
        progressDialog = SpotsDialog(this, R.style.DialogCustom)
        btMasuk.setOnClickListener {
            val username = etUsername.text.toString()
            val password = etPassword.text.toString()
            if (username.isEmpty() || password.isEmpty()) {
               val snackbar:Snackbar = Snackbar.make(coordinatorLogin,"Form Tidak Boleh Kosong",Snackbar.LENGTH_LONG)
                snackbar.setAction("MENGERTI",
                        View.OnClickListener {
                            snackbar.dismiss()
                        })
                val view:View = snackbar.view
                view.setBackgroundColor(Color.YELLOW)
                val tv:TextView = view.findViewById(com.google.android.material.R.id.snackbar_text)
                tv.setTextColor(Color.BLUE)
                snackbar.show()
            } else {
                progressDialog.show()
                loginPresenter.gotoLogin(username, password)
            }
        }

    }

    private fun initPresenter() {
        loginPresenter = LoginPresenter(this)
    }


    override fun _onLogin(loginResponse: LoginResponse) {
        progressDialog.dismiss()
        SharedPrefUtil.saveBoolean("isLogin", true)
        SharedPrefUtil.saveString("nama", loginResponse.jsonMember0!!.mpgnamagelar!!)
        SharedPrefUtil.saveString("password",loginResponse.jsonMember0!!.mpghandkey!!)
        SharedPrefUtil.saveString("kode_pegawai", loginResponse.jsonMember0!!.mpgkode!!)
        SharedPrefUtil.saveString("kode_unit", loginResponse.jsonMember0!!.msukode!!)
        SharedPrefUtil.saveString("nip", loginResponse.jsonMember0!!.mpgnip!!)
        startActivity(Intent(this@LoginActivity, MainActivity::class.java))
        finish()
    }

    override fun _onFailed() {
        progressDialog.dismiss()
        val snackbar = Snackbar.make(coordinatorLogin,"Login gagal",Snackbar.LENGTH_LONG)
        snackbar.setAction("MENGERTI",
                View.OnClickListener {
                    snackbar.dismiss()
                })
        val view:View = snackbar.view
        view.setBackgroundColor(Color.YELLOW)
        val textView:TextView = view.findViewById(com.google.android.material.R.id.snackbar_text)
        textView.setTextColor(Color.BLUE)
        snackbar.show()
    }
}
